src/Functions/formatReadableTimeDelta.cpp

```diff
@@ -172,39 +275,67 @@ class FunctionFormatReadableTimeDelta : public IFunction
     }
 
     static void processUnit(
-        UInt64 unit_size, const char * unit_name, size_t unit_name_size,
-        Float64 & value, WriteBuffer & buf_to, bool & has_output)
+        UInt64 unit_multiplier, UInt64 unit_divisor, const char * unit_name, size_t unit_name_size,
+        Float64 & value, WriteBuffer & buf_to, bool & has_output, Unit min_unit, bool is_minimum_unit)
```

Comment:
IMO using `Int128 & value` might eliminate the 'bad' cases caused by inexact representation of decimal values in floats.

Reply:
How do we represent non-whole numbers as Int128?

This function takes values like `formatReadableTimeDelta(123.123)`

Reply:
Well, you can multiply by the unit_multiplier of the mininum_unit

Reply:
Yes, tried it. Didn't help much - the number is a Float64 initially. There will always be a noticeable error